src/pageEditor/store/editor/editorSelectors/editorModSelectors.ts

```diff
@@ -0,0 +1,263 @@
+
+    const activatedModMetadatas: Array<ModComponentBase["modMetadata"]> =
+      modInstances.map((x) => mapModDefinitionToModMetadata(x.definition));
+
+    const baseMetadatas = uniqBy(
+      [...formStateModMetadatas, ...activatedModMetadatas],
+      (x) => x.id,
+    );
+
+    return baseMetadatas.map((metadata) => {
+      const dirtyMetadata = dirtyModMetadataById[metadata.id];
+      if (dirtyMetadata) {
+        return {
+          ...metadata,
+          ...dirtyMetadata,
+        } as ModMetadata;
+      }
+
+      return metadata;
+    });
+  },
+);
+
+export const selectModMetadataMap = createSelector(
+  selectActivatedModMetadatas,
+  (metadatas) => {
+    const metadataMap = new Map<RegistryId, ModMetadata>();
+    for (const metadata of metadatas) {
+      metadataMap.set(metadata.id, metadata);
+    }
+
+    return metadataMap;
+  },
+);
+
+const dirtyMetadataForModIdSelector = createSelector(
+  selectDirtyModMetadata,
+  (_state: EditorRootState, modId: RegistryId) => modId,
+  (dirtyModMetadataById, modId) =>
+    // eslint-disable-next-line security/detect-object-injection -- modId is a controlled string
+    dirtyModMetadataById[modId],
+);
+
+export const selectDirtyMetadataForModId =
+  (modId: RegistryId | undefined) => (state: EditorRootState) =>
+    modId ? dirtyMetadataForModIdSelector(state, modId) : null;
+
+///
+/// MOD OPTION DEFINITIONS
+///
+
+export const selectDirtyModOptionsDefinitions = ({ editor }: EditorRootState) =>
+  editor.dirtyModOptionsById;
+
+const dirtyOptionsDefinitionsForModIdSelector = createSelector(
+  selectDirtyModOptionsDefinitions,
+  (_state: EditorRootState, modId: RegistryId | null) => modId,
+  (dirtyOptionsDefinitionsByModId, modId) => {
+    if (modId == null) {
+      return;
+    }
+
+    // eslint-disable-next-line security/detect-object-injection -- RegistryId for mod
+    const options = dirtyOptionsDefinitionsByModId[modId];
+
+    if (options) {
+      // Provide a consistent shape of the options
+      return normalizeModOptionsDefinition(options);
+    }
+
+    // Return undefined if the options aren't dirty. Returning nullish instead of a default empty options allows the
+    // caller to distinguish no dirty options vs. options that have been reverted to the default.
+  },
+);
+
+export const selectDirtyOptionsDefinitionsForModId =
+  (modId: RegistryId | null) => (state: EditorRootState) =>
+    dirtyOptionsDefinitionsForModIdSelector(state, modId);
+
+///
+/// MOD OPTIONS ARGS
+///
+
+const dirtyOptionsArgsForModIdSelector = createSelector(
+  selectNotDeletedModComponentFormStates,
+  (_state: EditorRootState, modId: RegistryId | null) => modId,
+  (formStates, modId) =>
+    formStates.find((formState) => formState.modMetadata.id === modId)
+      ?.optionsArgs,
+);
+
+export const selectDirtyOptionsArgsForModId =
+  (modId: RegistryId | null) => (state: EditorRootState) =>
+    dirtyOptionsArgsForModIdSelector(state, modId);
```

Comment:
NIT: Should we be rewriting this as:

```suggestion
export const selectGetDirtyOptionsDefinitionsForModId = createSelector(
  (state: EditorRootState) => state,
  (state) =>
    memoize((modId: RegistryId | null) =>
      dirtyOptionsDefinitionsForModIdSelector(state, modId),
    ),
);
```

Reply:
Yep, I called that out in future work: "Standardize selecting data associated with a specific entity". I don't know the tradeoffs in ergonomics/performance at the callsite